_posts/2017-11-15-code-review.markdown

```diff
@@ -0,0 +1,32 @@
+---
+layout: post
+title:  "how to do code review, so it rocks"
+date:   2017-11-15 12:00:00 -0400
+categories: culture code-review
+author: Janice Smith
+---
+
+At Forward Financing we do a lot of code review. It's helped our team commit to delivering quality software and it's turned out pretty great. However, at first it can be hard to know how to dive in,
+so I'm going to shed some light on how it can be approached.
+
+In general the practice of code review helps peers collaborate on features. That collaboration brings fresh perspectives, exposing the coder and reviewer to new ways of approaching a problem they may not have thought of, while distributing that knowledge across the team. In order for this process to be successful, the review needs to be done well.
+
+Initially the coder should consider adding some context to a given PR. I like to put a description  of the changes and the feature, but it can also be nice to comment on specific lines that might be complex.  Also a link to the card or ticket can be a big help. When doing this, it is a good time for the coder to think about the PR as a whole.
+- Should this be more than one PR?
+- Should the code I write have any comments that to clarify or document any of these changes?
+
+Once the PR is ready to be reviewed, it can be good to ask for the opinion of specific engineers while still opening it up to get more points of view.  It can be great to get the eyes of someone who has spent a lot of time in the codebase or with that feature that is being updated, but just keep in mind as the coder, you will get different kinds of feedback based on who is reviewing. I like to try to get a few different perspectives if I can.
+
+As a reviewer, the first pass is a good time to look for areas of the code that jump out at you.
+- Are there any clearly unaddressed edge cases?
+- Are there any Deviations from the agreed style guide?
+
+Be sure to add notes to the PR in a clear and good natured way.
+
+Beyond that, a reviewer can have more specific questions that may be better served by a quick conversation, rather than a back and forth on the PR. It can be a big help on a team to get to know each other and how each person likes to communicate. Asking questions is also a good way to learn about the coder’s style or approach to solving an issue, “Why do you prefer a case statement here?”.  These kinds of questions can touch on performance, readability, extensibility. If it is a large feature or a big bug fix, it is great in code review to ask questions about where else this code is used or will be used. What may happen if the new code is used in ways maybe not intended by the coder? A conversion is one of the great things to come out of a code review.
```

Comment:
served by `having` or served ~by~ `with`

Reply:
Using active voice here would make this sentence/point a lot stronger:

`Learning how each person likes to communicate and having everyone get to know each other can be a big help for the team.`